src/math/bvh.js

```diff
@@ -0,0 +1,451 @@
+import { Vec3 } from '../math/vec3.js';
+import { BoundingBox } from '../shape/bounding-box.js';
+
+/** @typedef {import('../shape/ray.js').Ray} Ray */
+
+const INFINITY = Infinity;
+const BINS = 10;
```

Comment:
`BINS` should be configurable or a default property and `INFINITY` just doesn't helps anything